server/socket_server.go

```diff
@@ -84,16 +118,20 @@ func (s *SocketServer) acceptConnectionsRoutine() {
 		go func() {
 			// Wait until signal to close connection
 			errClose := <-closeConn
-			if errClose != nil {
-				fmt.Printf("Connection error: %v\n", errClose)
+			if err == io.EOF {
+				log.Warn("Connection was closed by client")
+				return // is this correct? the conn is closed?
```

Comment:
I think we should still call rmConn.  I guess it's ok to close a closed conn anyways.